src/fastoad/io/variable_io.py

```diff
@@ -11,34 +11,56 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import logging
 from fnmatch import fnmatchcase
-from os.path import exists, isfile
-from typing import IO, List, Sequence, Union
+from io import IOBase
+from os import PathLike
+from pathlib import Path
+from typing import List, Sequence, Union, IO, Optional
 
 from fastoad.openmdao.variables import VariableList
 from . import IVariableIOFormatter
 from .xml import VariableXmlStandardFormatter
+from .._utils.files import as_path
 from ..exceptions import FastError
 
+_LOGGER = logging.getLogger(__name__)  # Logger for this module
+
 
 class VariableIO:
     """
     Class for reading and writing variable values from/to file.
 
     The file format is defined by the class provided as `formatter` argument.
 
-    :param data_source: the I/O stream, or a file path, used for reading or writing data
+    :param data_source: I/O stream, or file path, used for reading or writing data
     :param formatter: a class that determines the file format to be used. Defaults to a
                       VariableBasicXmlFormatter instance.
     """
 
-    def __init__(self, data_source: Union[str, IO], formatter: IVariableIOFormatter = None):
+    def __init__(
+        self,
+        data_source: Optional[Union[str, PathLike, IO, IOBase]],
+        formatter: IVariableIOFormatter = None,
+    ):
+        if isinstance(data_source, (str, PathLike)):
+            data_source = as_path(data_source)
```

Comment:
Not really a remark, but a question on the definition of the `as_path` method. Why are there 3 definitions with an `@overload` decorator

Reply:
And now a remark: this method does not change the argument if it's already a path so why include the `PathLike` type in the if ?

Reply:
> Not really a remark, but a question on the definition of the `as_path` method. Why are there 3 definitions with an `@overload` decorator
It allows to specify different signatures. Here, it allows to have different return types according to the input type.
It is somehow overkill, but I wanted to test this feature :)

Reply:
> And now a remark: this method does not change the argument if it's already a path so why include the `PathLike` type in the if ?
Strictly speaking, a PathLike object can be something else than a Path object. It is just guaranteed that a Path object can be built from a PathLike object.

On a purely pragmatic point of view, to avoid the `if` statement, `as_path()` could return the original object if it cannot be converted to a Path object. Currently, in such case, it returns None.
The current way seems a more intuitive behavior to me, given the name of the function, so I'm still undecided...